contributing.md

```diff
@@ -119,11 +119,19 @@ make docs
 
 ## Releasing
 
+Our current process for publishing a release of python-pachyderm consists of the following steps:
+
+- [ ] Ensure compatibility with the latest protos:
+  - [ ] Make sure `version.json` references the most recent compatible release of Pachyderm
+  - [ ] Run `make src/python_pachyderm/proto/v2` to pull the latest pachyderm code
+  - [ ] Run `make lint` and our test suite (see `contributing.md`)
+- [ ] Ensure that all examples still pass:
+  - [ ] All examples in the `examples/` dir (they must be run manually—`tox example` only runs `examples/opencv`
+  - [ ] [Pachyerm's spouts101 example](github.com/pachyderm/pachyderm/tree/master/examples/spouts101) (should match `examples/spouts101` in this repo, but it's good to confirm
+- [ ] Rebuild docs to make sure they're in sync, and push the updated docs
+- [ ] Update `version.json` to the next version of `python-pachyderm` (this will be the released version). Commit and push this change to `master`
```

Comment:
I added a Makefile command called `test-release` which uploads the package to test-pypi website. Might be good to upload here, make sure everything looks good (from the readme to the compatible python versions etc.) before publishing to pypi because pypi doesn't allow you to re-publish the same version. This means if we mess up, we're forced to bump up the version.

Reply:
Oh, awesome! Added this